lib/factorial.rb

```diff
@@ -1,6 +1,17 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: Linear or O(n), where n stands for input.  However many times the loop runs depends on input.
+# Space complexity: Constant 1(n), where n stands for input.  Additional space doesn't rely on input.
 def factorial(number)
-  raise NotImplementedError
+  raise ArgumentError if number == nil
+
+  i = 2
```

Comment:
Interesting approach. i = 2 feels like 2 should be a well named constant, since right now there's no indication of why it is set to two.